src/master-playlist-controller.js

```diff
@@ -1296,8 +1306,12 @@ export class MasterPlaylistController extends videojs.EventTarget {
    *
    * @param {Object=} error an optional error that may include the playlist
    * to blacklist
+   * @param {Number=} blacklistDuration an optional number of seconds to blacklist the
+   * playlist
+   * @param {Boolean=} skipPlaylistSelection an optional boolean to skip playlist
+   * selection
    */
-  blacklistCurrentPlaylist(error = {}) {
+  blacklistCurrentPlaylist(error = {}, blacklistDuration, skipPlaylistSelection = false) {
```

Comment:
is `skipPlaylistSelection` really necessary? I think it would be better to leave playlist selection here as is and just trigger `abortearly` and not `bandwidthupdate` from `segmentLoader`. If we are blacklisting the current playlist, I don't really see a scenario in which we would not want to switch from it. I think it's also reasonable to not trigger `bandwidthupdate` for the early abort case, as the `bandwidth` set in the segment loader after an early abort is not necessarily the bandwidth of the user, but a specific bandwidth to force playlist selection to select a certain variant. It's a bit of a lie for segment loader to claim it has an update to calculated bandwidth.